[1.12][cherrypick] use old namespace in resource modifier (#6724)

anshulahuja98 opened the pull request:
* use old namespace in resource modifier



* add changelog



* update docs



* updated after review



---------

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

ywk253100 commented:
@anshulahuja98 The original PR is out of data with the base branch, so I rebased it by clicking a button. But the DCO checking doesn't pass now, could you take a look at this?

anshulahuja98 commented:
> @anshulahuja98 The original PR is out of data with the base branch, so I rebased it by clicking a button. But the DCO checking doesn't pass now, could you take a look at this?

Fixed now.
